src/App.js

```diff
@@ -165,19 +162,32 @@ function App() {
     [guesses, word]
   );
 
+  const latestMatchCounts = useMemo(() => {
+    const allResults = letters.map((l) => matches[l]);
+    const newMatches = allResults.filter(x => x==="match").length - foundMatches;
+    setFoundMatches(foundMatches + newMatches);
```

Comment:
We shouldn't be calling `setFoundMatches()` from within a `useMemo`. `useMemo` should have no side effects and should just perform the calculations for `latestMatchCounts`.

Reply:
eliminated `const latestMatchCounts`.